framework/include/restart/Restartable.h

```diff
@@ -72,6 +73,19 @@ class Restartable
   template <typename T, typename... Args>
   T & declareRestartableData(const std::string & data_name, Args &&... args);
 
+  /**
+   * Declare a piece of data as "restartable" and initialize it
+   * Similar to `declareRestartableData` but returns a const reference to the object.
+   *
+   * NOTE: This returns a _ const reference_!  Make sure you store it in a _const reference_!
+   *
+   * @param data_name The name of the data (usually just use the same name as the member variable)
+   * @param args Arguments to forward to the constructor of the data
+   */
+  template <typename T, typename... Args>
+  const T & getRestartableData(const std::string & data_name, Args &&... args) const;
```

Comment:
Do we need to have constructor arguments for a getter?

I would think that if we have an object that is getting something but not declaring it, it must be trivial enough to have a default constructor. Even if this is in a restart where nothing ever declares it, and we're only getting